meta-cube/recipes-core/images/cube-essential_0.1.bb

```diff
@@ -42,5 +42,68 @@ INITRAMFS_IMAGE = "cube-builder-initramfs"
 # We want it separate, and not bundled with the kernel by default.
 INITRAMFS_IMAGE_BUNDLE ?= ""
 
+IMAGE_FEATURES += "read-only-rootfs"
+
 inherit core-image
 inherit builder-base
+
+ROOTFS_POSTPROCESS_COMMAND += '${@bb.utils.contains("IMAGE_FEATURES", "read-only-rootfs", "read_only_essential; ", "",d)}'
+
+read_only_essential () {
+    echo "none	/etc/openvswitch	tmpfs	defaults	1	1" >> ${IMAGE_ROOTFS}/etc/fstab
+    mkdir -p ${IMAGE_ROOTFS}/opt/container
+    echo "none	/opt/container	tmpfs	defaults	1	1" >> ${IMAGE_ROOTFS}/etc/fstab
+    echo "none	/var/lib/misc	tmpfs	defaults	1	1" >> ${IMAGE_ROOTFS}/etc/fstab
+    if [ -e ${IMAGE_ROOTFS}/etc/hosts ]; then
+        mv ${IMAGE_ROOTFS}/etc/hosts ${IMAGE_ROOTFS}/etc/hosts0
+        ln -s ../run/systemd/resolve/hosts ${IMAGE_ROOTFS}/etc/hosts
```

Comment:
A general question about the linked files to where they can be written.

What changes will be lost on reboot ? We should make that clear in a README, i.e. just a paragraph that describes what read-only essential means if you reboot the system "any changes to the /etc/* files will be lost", etc.